### ISSUE TITLE
@mantine/dates setting fixOnBlur to DatePicker has no effect on onChange calls 

### ISSUE THREAD

**Issue by Foohx** [mantinedev/mantine#532]
### What package has an issue

@mantine/dates

### Describe the bug

When the `fixOnBlur` prop is set to `false` on `DatePicker` component. The `onChange` method is still called when the field is unfocused.

This can produce random outcomes on complex dynamic forms. In NextJs project, the `onChange` method is called multiple times causing a lot of bugs.

### In which browser did the problem occur

All

### If possible, please include a link to a codesandbox with the reproduced problem

React : https://codesandbox.io/s/strange-satoshi-b0uz9?file=/src/App.tsx
NextJs : https://codesandbox.io/s/next-datepicker-cmi67?file=/pages/index.tsx

As you can see, it's randomly impossible to clear the input on a simple Next project.

### Do you know how to fix the issue

No

**Comment by Foohx** [comment]
After some tests, it seems that forcing `value` prop to be `null` (example below) corrects the problem with `fixOnBlur`.

The temporary solution :
```tsx
// value : Date | null

 <DatePicker
    fixOnBlur={false}
    // ...
    value={value === null ? null! : value}
    onChange={(newValue) => {
        console.log("OLD", value, "NEW", newValue);
        onChange(newValue);
    }}
 />
```

In `DatePickerProps` interface of DatePicker component, the `value` property is typed as `Date | undefined`. Maybe adding null might be a good thing.

```ts
/** Selected date, required with controlled input */
value?: Date | null;
```

**Comment by rtivital** [comment]
Fixed in 3.4.3
